ffmpeg/ffmpeg_test.go

```diff
@@ -1512,3 +1512,60 @@ func TestTranscoder_FormatOptions(t *testing.T) {
 		t.Error("Did not get expected error with invalid format ", err)
 	}
 }
+
+// test short segments
+func shortSegments(t *testing.T, accel Acceleration, fc int) {
+	run, dir := setupTest(t)
+	defer os.RemoveAll(dir)
+
+	cmd := `
+    # generate segments with fc-frames
+    cp "$1/../transcoder/test.ts" .
+	frame_count=%d
+
+    ffmpeg -loglevel warning -ss 0 -i test.ts -c copy -frames:v $frame_count -copyts short0.ts
+    ffmpeg -loglevel warning -ss 2 -i test.ts -c copy -frames:v $frame_count -copyts short1.ts
+    ffmpeg -loglevel warning -ss 4 -i test.ts -c copy -frames:v $frame_count -copyts short2.ts
+    ffmpeg -loglevel warning -ss 6 -i test.ts -c copy -frames:v $frame_count -copyts short3.ts
+
+    ffprobe -loglevel warning -count_frames -show_streams -select_streams v short0.ts | grep nb_read_frames=$frame_count
+    ffprobe -loglevel warning -count_frames -show_streams -select_streams v short1.ts | grep nb_read_frames=$frame_count
+    ffprobe -loglevel warning -count_frames -show_streams -select_streams v short2.ts | grep nb_read_frames=$frame_count
+    ffprobe -loglevel warning -count_frames -show_streams -select_streams v short3.ts | grep nb_read_frames=$frame_count
+  `
+	run(fmt.Sprintf(cmd, fc))
+
+	tc := NewTranscoder()
+	defer tc.StopTranscoder()
+	for i := 0; i < 4; i++ {
+		fname := fmt.Sprintf("%s/short%d.ts", dir, i)
+		t.Log("fname ", fname)
+		in := &TranscodeOptionsIn{Fname: fname, Accel: accel}
+		out := []TranscodeOptions{{Oname: dir + "/out.ts", Profile: P144p30fps16x9, Accel: accel}}
+		res, err := tc.Transcode(in, out)
+		if err != nil {
+			t.Error(err)
+		}
+		if fc != res.Decoded.Frames {
+			t.Error("Did not decode expected number of frames: ", res.Decoded.Frames)
+		}
+		if 0 == res.Encoded[0].Frames {
```

Comment:
Is there a predictable pattern to the number of encoded frames that we can check against here?

Reply:
they do belong in `(decoded frames/2) ± 1` will that be a better check along with checking for zero?

Reply:
> I guess only checking for non-zero frames should be good enough for now? 

Yep, good enough for now. Correct me if I'm wrong, but ffmpeg also fails to encode anything for the few-frame cases, so we can't easily sanity check these cases with ffmpeg cli?

Reply:
Yep ffmpeg fails for our 1-frame case unless we're using `eof_action=pass`. I'll test if we can possibly compare with that.

Reply:
Any luck in comparing some of the cases  with ffmpeg CLI and `eof_action=pass` ?

Reply:
Nope they varied by +-1 for almost half the non-single-frame cases